### ISSUE TITLE
macOS Sonoma Support

### ISSUE THREAD

**Issue by ghsssss** [OpenIntelWireless/itlwm#883]

**Comment by zxystd** [comment]
等。itlwm.kext先顶着。

**Comment by qcTimb3rlandqc** [comment]
> Apple had deleted the whole legacy 80211 stack on Sonoma, I am going to turn to the other `IO80211Family` which is using by macintosh of late 2018(although it will be also deleted in 2~3 years), it is a big work, almost all of the airport interfaces and network interfaces changed, need to start reversed over again. Just wait, I don't have too much spare time to diverse.
> 
> ![image](https://user-images.githubusercontent.com/16227538/244144097-15737ebc-0bc4-4682-aaf8-bf98a222602c.png)
> 
> You can use `itlwm.kext + Heliport.app` instead for the while.

if you need tester lets me no ;)thx for work perfect job :)

**Comment by ameenjuz** [comment]
@zxystd 
any progress AirportItlwm.kext for Sonoma support?
I missed this kext for Sonoma and I m waiting

**Comment by zxystd** [comment]
New IO80211Family is loaded. but it is just a beginning...
![ce93852c37fe7f85390e17067172dcae](https://github.com/OpenIntelWireless/itlwm/assets/16227538/0be2d9af-2e07-4289-86a6-d75efbac080a)

**Comment by cheesycod** [comment]
is there anything newer than the 2018 80211?

**Comment by zxystd** [comment]
![image](https://github.com/OpenIntelWireless/itlwm/assets/16227538/3ed82fb0-a891-4e75-a8e6-c684a62ae4b0)
Now network controller start success and begin to exchange informations.

The IOCTLs almost rewrite, hundreds of IOCTL structs should be reversed...

**Comment by cheesycod** [comment]
any chance of making a ath10k driver too or smth, just asking since I know that's a pipe dream

**Comment by laszlo-t** [comment]
> @zxystd any progress AirportItlwm.kext for Sonoma support? I missed this kext for Sonoma and I m waiting

the itlwm.kext works for me on Sonoma

**Comment by hendrilmendes** [comment]
> > @zxystd any progress AirportItlwm.kext for Sonoma support? I missed this kext for Sonoma and I m waiting
> 
> the itlwm.kext (v2.2.0) works for me on Sonoma

what model of your wifi card?

**Comment by laszlo-t** [comment]
AX201

Hendril Mendes ***@***.***> ezt írta (időpont: 2023.
jún. 16., P, 15:01):

**Comment by TheDarkosGang** [comment]
> > @zxystd any progress AirportItlwm.kext for Sonoma support? I missed this kext for Sonoma and I m waiting
> 
> the itlwm.kext (v2.2.0) works for me on Sonoma (AX201)

Did you use any boot-args?

**Comment by laszlo-t** [comment]
> > the itlwm.kext (v2.2.0) works for me on Sonoma (AX201)
> 
> Did you use any boot-args?

not for the wifi

**Comment by Lorys89** [comment]
> ![image](https://user-images.githubusercontent.com/16227538/245754639-3ed82fb0-a891-4e75-a8e6-c684a62ae4b0.png) Now network controller start success and begin to exchange informations.
> 
> The IOCTLs almost rewrite, hundreds of IOCTL structs should be reversed...

Thank you for your commitment, how is the development progressing?

**Comment by zxystd** [comment]
![8f384e66b1c13f0e0578880a5b85e70b](https://github.com/OpenIntelWireless/itlwm/assets/16227538/a3513db2-a637-4653-903a-d41b3946ee6e)
Now we get the wifi information.

**Comment by ameenjuz** [comment]
> ![8f384e66b1c13f0e0578880a5b85e70b](https://user-images.githubusercontent.com/16227538/247016841-a3513db2-a637-4653-903a-d41b3946ee6e.png) Now we get the wifi information.
Great
I m waiting your test kext

**Comment by RyzenDew** [comment]
Such fast work

**Comment by Lorys89** [comment]
> ![8f384e66b1c13f0e0578880a5b85e70b](https://user-images.githubusercontent.com/16227538/247016841-a3513db2-a637-4653-903a-d41b3946ee6e.png) Now we get the wifi information.

Have you already made the scan of wifi networks work?

**Comment by zxystd** [comment]
Now we get the wifi list.
![image](https://github.com/OpenIntelWireless/itlwm/assets/16227538/e590b992-24fe-417e-bdaf-4d25332009ff)

**Comment by Lorys89** [comment]
> Now we get the wifi list. ![image](https://user-images.githubusercontent.com/16227538/247445041-e590b992-24fe-417e-bdaf-4d25332009ff.png)

I'm speechless, you're really the Top! 
(if you need test I have 4 hackintosh with wifi cards 9462, 9560, ax200, ax210)

**Comment by Lorys89** [comment]
Today probably comes out beta 2 of sonoma, it would be nice to already start testing your kext 😝

**Comment by aryajuanda** [comment]
> Now we get the wifi list.
> ![image](https://user-images.githubusercontent.com/16227538/247445041-e590b992-24fe-417e-bdaf-4d25332009ff.png)

nice work... very good progress, 

BTW the PCI vendor is 0x15AD (Vmware) not 0x8086 (intel)

**Comment by RyzenDew** [comment]
> 

they are working within vmware is my guess but they are still doing good work none the less and yes you can put wifi and BT in to vmware and have it work i just tried it

**Comment by RyzenDew** [comment]
> Now we get the wifi list. ![image](https://user-images.githubusercontent.com/16227538/247445041-e590b992-24fe-417e-bdaf-4d25332009ff.png)

any updates on when we can test the kexts?

**Comment by ameenjuz** [comment]
@zxystd nice work good progress
 can you provide the kext for testing?

**Comment by xjh169** [comment]
坐等各位 发kext ，我的是 Wireless-AC 8265

**Comment by bilijp153** [comment]
坐等发kext，我的是 AC 9461

**Comment by notAperson535** [comment]
> Now we get the wifi list. ![image](https://user-images.githubusercontent.com/16227538/247445041-e590b992-24fe-417e-bdaf-4d25332009ff.png)

Wow, you're definitely making progress!

**Comment by notAperson535** [comment]
And to all the people asking to test the kext, usually whenever they commit the new changes the kext appears in releases as a prerelease. So just wait until they push the changes.

**Comment by zxystd** [comment]
Now I am stuck at how to submit network packets into the new `IO80211Family` network stack, different with `Apple Silicon WiFi drivers`, I don't want to dig deep into the `Skywalk` stack which will almost force me rewrite the whole driver including the HAL layer, so I am  faking the network packets going through the Ethernet interface instead of `IOSkywalkLegacyEthernet` and `IOSkywalkLegacyEthernetInterface`.

Yup, I am debugging on the Vmware with a Realtek USB wireless dongle, because Intel PCIE cards can not do passthrough on macOS. So the achieving results should be sync to AirportItlwm after the entire process running smoothly.
![90d16fe696ad67362485727f32834232](https://github.com/OpenIntelWireless/itlwm/assets/16227538/69438af3-5414-43a9-9679-9875d20e9f22)

**Comment by Lorys89** [comment]
How do you proceed with the problem of network packets? Did you fix this problem?

**Comment by o3p6** [comment]
Cheer on

**Comment by zxystd** [comment]
Good news! Now we can connect to the internet and have a smoothly network, also have the two problem to be resolved.
- Can not DHCP, can be used by manual setting IP address.
- Can not load by OpenCore, can be loaded by manual load.

![qq_pic_merged_1688054906091](https://github.com/OpenIntelWireless/itlwm/assets/16227538/67d3c654-84bd-4174-abd2-c14fc10a689f)
![qq_pic_merged_1688055228093](https://github.com/OpenIntelWireless/itlwm/assets/16227538/e8bf9741-d68c-4057-be25-439b596b4b57)

**Comment by zxystd** [comment]
For the loading problem, here is the error log, I don't know if it is the error of my kext's virtual table declaration or the OpenCore kext injection, anyone have ideas? Is the SkywalkFamily located in the prelinkedKC?

![image](https://github.com/OpenIntelWireless/itlwm/assets/16227538/db29dafa-5f84-490d-a80b-4edb87f3293e)
[opencore-2023-06-30-041658.txt](https://github.com/OpenIntelWireless/itlwm/files/11915440/opencore-2023-06-30-041658.txt)

**Comment by hendrilmendes** [comment]
I'm using opencore 0.9.4, I can test to see if the same thing happens

**Comment by hendrilmendes** [comment]
Sonoma not working🥹

**Comment by khaled-muhammad** [comment]
I want to know the needed skills to do something like this

**Comment by zxystd** [comment]
New Progress!
Loading issue resolved, thanks for @imcdd inspired me.
The last problem is the IPv4 DHCP, will see it after weekend.

**Comment by Lorys89** [comment]
@zxystd you have succeeded in this great undertaking, but will you be able to make airdrop work on intel cards?

**Comment by khaled-muhammad** [comment]
@zxystd Can you please tell me what to learn to make such a driver or thing, so that I can help.

**Comment by zxystd** [comment]
DHCP works!
![42569a6fffe46e16eed79a4c545ecbbb](https://github.com/OpenIntelWireless/itlwm/assets/16227538/098571aa-d99b-4b89-8b15-8678cf238e84)
![36c07f117142eca1dd03de03d309b3ff](https://github.com/OpenIntelWireless/itlwm/assets/16227538/c2e8ef2d-496c-4022-8592-6f00586706f1)

**Comment by rchunping** [comment]
> DHCP works!
> ![42569a6fffe46e16eed79a4c545ecbbb](https://user-images.githubusercontent.com/16227538/250454919-098571aa-d99b-4b89-8b15-8678cf238e84.png)
> ![36c07f117142eca1dd03de03d309b3ff](https://user-images.githubusercontent.com/16227538/250455000-c2e8ef2d-496c-4022-8592-6f00586706f1.png)

great！

**Comment by Daasin** [comment]
@zxystd Forgive me if this counts as off-topic but regarding the changes you made to be on skywalk instead of legacy family. Would that impact stuff like handover & continuity? 
(I know you said you disabled handover by default as of ventura  but I'm hoping there's a boot-arg flag that we can set if we don't mind less speed and more instability to get closer to native features 🤞🏽) 

And in anycase, thank you for your efforts on this, it truly makes a huge impact on the community and we are very grateful for your work. 🙏🏽

**Comment by o3p6** [comment]
> 

？

**Comment by zxystd** [comment]
> @zxystd Forgive me if this counts as off-topic but regarding the changes you made to be on IO80211Family (or skywalk) instead of the legacy family. Would that impact stuff like handover & continuity?
> 

It won't have an impact, I spoof the skywalk transport layer, not the upper controller layer, so it is also a airport network controller.

> (I know you said you disabled handover by default as of ventura but I'm hoping there's a boot-arg flag that we can set if we don't mind less speed and more instability to get closer to native features 🤞🏽)
> 

The continuity feature may be reenable in Sonoma version later, and maybe Ventura also have the chance to use IO80211Family instead of IO80211FamilyLegacy.

> And in anycase, thank you for your efforts on this, it truly makes a huge impact on the community and we are very grateful for your work. 🙏🏽

Thank you, it definitely takes me much time...

**Comment by ExtremeXT** [comment]
Why not port all versions to IO80211Family (or IO80211FamilyV2)?

**Comment by qcTimb3rlandqc** [comment]
> Great job, when the release of this version to start with testing?

Never for asking personne

**Comment by zxystd** [comment]
[AirportItlwm-Sonoma-Preview01.zip](https://github.com/OpenIntelWireless/itlwm/files/11944708/AirportItlwm-Sonoma-Preview01.zip)
The first preview version.

**Comment by Lorys89** [comment]
> [AirportItlwm-Sonoma-Preview01.zip](https://github.com/OpenIntelWireless/itlwm/files/11944708/AirportItlwm-Sonoma-Preview01.zip) The first preview version.

Works great, you're the top💪

**Comment by maxpicelli** [comment]
Thanks a lot my friend!!! Works fine with AX211 intel!!! It's possible that works Airdrop? I using Fenvi T-1919 together with WIFI6 native from MOBO Gigabyte Aorus Z790. If works Airdrop im retired Fenvi Pcie.

**Comment by neresident** [comment]
> [AirportItlwm-Sonoma-Preview01.zip](https://github.com/OpenIntelWireless/itlwm/files/11944708/AirportItlwm-Sonoma-Preview01.zip) The first preview version.

Great work, @zxystd! I wanted to clarify, will there be a similar test update for Ventura / Monterey and other early systems? Ofc, if it's possible to release that for you. Thanks in advance!

**Comment by sytao123** [comment]
Thank you for your hard work！I will try it as soon as possible.

**Comment by alcheng3344** [comment]
@zxystd Thank you for your Great work on AirportItlwm，it works now 
![截圖 2023-07-04 下午3 06 54](https://github.com/OpenIntelWireless/itlwm/assets/94786643/f1f61f03-caa4-4054-b4a2-693c9d383788)

**Comment by nsby494** [comment]
非常不错，这个必须要汉字留言！！

**Comment by 906051999** [comment]
nice

**Comment by lwx1901** [comment]
Z大，您好，首先感谢你作出的重大贡献，测试中发现一个问题：就是Hackintool检测到无线网卡没有内建？不知道是个案还是普遍问题。
![截屏2023-07-04 23 23 57](https://github.com/OpenIntelWireless/itlwm/assets/81496981/fb3a87c4-8169-4b5c-8277-d2d9b7c412b5)

**Comment by szp88** [comment]
是的。军刀里显示没有内建√是没有打上的。



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2023年07月04日 23:24 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [OpenIntelWireless/itlwm] macOS Sonoma Support (Issue #883) |

Z大，您好，首先感谢你作出的重大贡献，测试中发现一个问题：就是Hackintool检测到无线网卡没有内建？不知道是个案还是普遍问题。

**Comment by maxpicelli** [comment]
> Z大，您好，首先感谢你作出的重大贡献，测试中发现一个问题：就是Hackintool检测到无线网卡没有内建？不知道是个案还是普遍问题。 ![截屏2023-07-04 23 23 57](https://user-images.githubusercontent.com/81496981/250943258-fb3a87c4-8169-4b5c-8277-d2d9b7c412b5.png)

It's possible edit this with plist. extracted from Hackintool? 

Mine show up this:
![Captura de Tela 2023-07-04 às 12 48 23](https://github.com/OpenIntelWireless/itlwm/assets/115906479/78c72dc7-1183-4837-bc3e-5e5e6501bbe3)

**Comment by bilijp153** [comment]
![截屏2023-07-04 23 50 12](https://github.com/OpenIntelWireless/itlwm/assets/86920625/a3168828-00e8-4420-9742-a7a055ab9021)
一样，我的也没有内建

**Comment by bilijp153** [comment]
![截屏2023-07-04 23 55 39](https://github.com/OpenIntelWireless/itlwm/assets/86920625/2b42f499-0e53-48f4-88ba-0196b18c16a0)
系统属性里还是能瞧见的

**Comment by al5710** [comment]
ax210 work

**Comment by Kaisar870** [comment]
Can anybody help with the kext? I've put the preview kext on my kext folder and changed the plist like the attached screenshot. Do I need anything else?
![Screenshot 2023-07-05 002836](https://github.com/OpenIntelWireless/itlwm/assets/75267509/718b0b05-a186-4629-aed6-8a2124025de5)

**Comment by Kaisar870** [comment]
> 

can you help me with that? i'm trying to fix my wifi but seems like i'm doing something wrong

**Comment by Kaisar870** [comment]
> ax210 work

can you help me with that? i'm trying to fix my wifi but seems like i'm doing something wrong

**Comment by 553051957** [comment]
> [AirportItlwm-Sonoma-Preview01.zip](https://github.com/OpenIntelWireless/itlwm/files/11944708/AirportItlwm-Sonoma-Preview01.zip) The first preview version.
> 
> **Considering this issue is for tracking the dev progress, if you have any issue on the kext or have any suggestions, please go to this issue #896.**

反馈一个bug,两个驱动共同存在问题，都是无法连接成功带wp3协意的网络，协意改成wp2就是完美连接

**Comment by Eternity5510** [comment]
Its works fine =)
![77](https://github.com/OpenIntelWireless/itlwm/assets/51543953/1c9a481d-062e-4282-bb53-218564ac8ecc)

**Comment by tyu666888** [comment]
It seems that you can only connect to the 5G network, 2.4G is recognized, and the connection will not be able to connect after 1 second, do you need to manually set the IP

**Comment by joostiphone** [comment]
I've loaded the kext into EFI-OC-Kexts and set MinKernel to 23.0.0 and MaxKernel to 23.99.99, but it doesn't load. Any advise?

**Comment by maxpicelli** [comment]
What kext are you using to put this configuration? If you use Broadcom Airport kext not needed put in OC. Bluetooth recognize Native. And to Sonoma not need this. If you are using for Broadcom without Airport intel need 22.99.99 MAxKernel and 12.0.0 for use AirportBrcmFixup.kext. Not need for AirportItlwm.kext.

**Comment by maxpicelli** [comment]
> ![Screenshot 2023-07-05 002836](https://user-images.githubusercontent.com/75267509/250972218-954c6c2b-0543-4881-b7c2-2a53e7d2a2e9.png)
> 
> > ax210 work
> 
> can you help me with that? i'm trying to fix my wifi but seems like i'm doing something wrong

What you try using deprecate kexts on Sonoma? Need more informations about kexts in OC.

**Comment by Lorys89** [comment]
@zxystd please fix the device builtin
![Screenshot 2023-07-05 alle 20 24 14](https://github.com/OpenIntelWireless/itlwm/assets/64613690/a439fbe6-02ba-4152-b132-0bdd7d2ef9bd)

without builtin not work imessage
I have already tried to inject the built-in wifi 00 and 01 and it doesn't work

**Comment by joostiphone** [comment]
> What kext are you using to put this configuration? If you use Broadcom Airport kext not needed put in OC. Bluetooth recognize Native. And to Sonoma need, If you are using for Broadcom without Airport intel, is set 22.99.99 MaxKernel and 12.0.0 MinKernel for use AirportBrcmFixup.kext. Not need setup for AirportItlwm.kext.

Please see attached screenshot and[ here my EFI.](https://mega.nz/file/RtAlSAiA#4RnMna4qnhIqi7J-V_5yZSgr2VIWeH_hGcqi83nFgoo) 

<img width="1304" alt="2023-07-05_21-26-10" src="https://github.com/OpenIntelWireless/itlwm/assets/16277235/6565cc6b-1406-4d59-a2c5-b713d2bd66d7">

Can you do the changes, and uploaded it again?

**Comment by maxpicelli** [comment]
[config.zip](https://github.com/OpenIntelWireless/itlwm/files/11962850/config.zip)

Just replace your config.plist. At the same without 23.0.0 and 23.99.99 because Airport not need it.

**Comment by zxystd** [comment]
Update
```
Add 'built-in' properties.
```

Known issue:

- WPA/WPA2 enterprise network can not be connected because Apple deleted the `RSNSupplicant`, since Broadcom devices handling this feature in there firmwares side. Will not consider to fix until I find solutions.
- We had switched to itlwm build-in supplicant to handle cipher keys handshake, so if the Wi-Fi password the user input is wrong, it may not remind the user.

I am recover busy state for these days.

[AirportItlwm-Sonoma-Preview02.zip](https://github.com/OpenIntelWireless/itlwm/files/11980729/AirportItlwm-Sonoma-Preview02.zip)

**Comment by zxystd** [comment]
Update
```
Limit WPA WPA2 or MIXED networks can be connected, clean other unsupported flags.
```

[AirportItlwm-Sonoma-Preview03.zip](https://github.com/OpenIntelWireless/itlwm/files/12068076/AirportItlwm-Sonoma-Preview03.zip)

**Comment by zxystd** [comment]
Update 2023.7.18

Support `WPA/WPA2 Enterprise` network.

[AirportItlwm-Sonoma-Preview04.zip](https://github.com/OpenIntelWireless/itlwm/files/12079097/AirportItlwm-Sonoma-Preview04.zip)

**Comment by zxystd** [comment]
Update 2023.8.9

Support Sonoma beta 5+

It's the first time I've seen API interfaces modified in the same macOS major version.
**Don't use it on Sonoma beta 4 and below !!!**

[AirportItlwm-Sonoma-Preview05.zip](https://github.com/OpenIntelWireless/itlwm/files/12299048/AirportItlwm-Sonoma-Preview05.zip)

**Comment by zxystd** [comment]
Already merged into main branch 306562e4925bee0917ac367a091e305f63ab205e, now we can download from alpha version release page. Thanks for everyone.
